lighthouse-core/scoring.js

```diff
@@ -41,22 +48,20 @@ class ReportScoring {
 
       category.audits.forEach(audit => {
         const result = resultsByAuditId[audit.id];
-        // Cast to number to catch `null` and undefined when audits error
-        let auditScore = Number(result.score) || 0;
-        if (typeof result.score === 'boolean') {
-          auditScore = result.score ? 100 : 0;
-        }
         // If a result was not applicable, meaning its checks did not run against anything on
         // the page, force it's weight to 0. It will not count during the arithmeticMean() but
         // will still be included in the final report json and displayed in the report as
         // "Not Applicable".
         if (result.notApplicable) {
-          auditScore = 100;
+          result.score = 1;
           audit.weight = 0;
           result.informative = true;
         }
+        result.score = clamp2decimals(result.score);
 
-        result.score = auditScore;
+        if (!Number.isFinite(result.score)) {
```

Comment:
do we need to check in both places?

Reply:
i'd be fine with also doing this in generateAuditResult, however IMO there's not a big diff. 

* Many of our tests assert the result returned by `audit()`
* `Audit.generateAuditResult` is called via Runner once each audit is done
* Basically right after that, we call this `scoreAllCategories` method.

So ideally we'd have something in Audit class that wraps each audit's `audit()` method, but we don't.

Reply:
lol I just made your point for you. 😊